### ISSUE TITLE
Webpack-Dev-Middleware errors for different Versions.

### ISSUE THREAD

**Issue by GuacheSuede** [webpack/webpack-dev-middleware#296]
* Operating System: Windows 10
* Node Version:  8.7.0
* NPM Version: YARN 1.3.2
* webpack version: 4.5.0
* webpack-dev-middleware Version: 3.1.2

- [ ] This is a **feature** request
- [x] This is a **bug**

### Code

```js
webpack.config.js
const path = require('path');
const HtmlWebpackPlugin = require('html-webpack-plugin');
const CleanWebpackPlugin = require('clean-webpack-plugin');

module.exports = {
  entry: {
    app: './src/index.js',
    print: './src/print.js'
  },
  devtool: 'inline-source-map',
  plugins: [
    new CleanWebpackPlugin(['dist']),
    new HtmlWebpackPlugin({
      title: 'Output Management'
    })
  ],
  output: {
    filename: '[name].bundle.js',
    path: path.resolve(__dirname, 'dist'),
     publicPath: '/'
  }
};
```
```js
server.js
const express = require('express');
const webpack = require('webpack');
const webpackDevMiddleware = require('webpack-dev-middleware');

const app = express();
const config = require('./webpack.config.js');
const compiler = webpack(config);

// Tell express to use the webpack-dev-middleware and use the webpack.config.js
// configuration file as a base.
app.use(webpackDevMiddleware(compiler, {
  publicPath: config.output.publicPath
}));

// Serve the files on port 3000.
app.listen(3000, function () {
  console.log('Example app listening on port 3000!\n');
});
```

### Expected Behavior
Everything Loads Smoothly as expected in Webpack Development Guide 

### Actual Behavior
Cannot GET / Error > 3.1.2
MIME Error `Refused to execute script from 'http://localhost:3000/app.bundle.js' because its MIME type ('text/html') is not executable, and strict MIME type checking is enabled.` < 3.0.1

### For Bugs; How can we reproduce the behavior?
I followed Webpack Guide on Development (https://webpack.js.org/guides/development/). Anything Below Version 3.0.1 fails with `MIME ERROR` and anything Above fails with `Cannot GET /` Error

**Comment by MitchTalmadgeUofU** [comment]
Thanks for writing this up. For reference, the issue was discovered here: https://github.com/webpack/webpack-dev-server/issues/1373

**Comment by GuacheSuede** [comment]
Wow, on the same day ! Would you rather I close this issue ?

**Comment by MitchTalmadgeUofU** [comment]
No, this is much closer to the source of the problem, let's keep both issues open :)

**Comment by shellscape** [comment]
@GuacheSuede @MitchTalmadgeUofU which browser and version are you using that you're seeing this? (I'd guess Chrome, but need to confirm)

**Comment by MitchTalmadgeUofU** [comment]
@shellscape Chrome Version 65.0.3325.181 (Official Build) (64-bit)

**Comment by shellscape** [comment]
@MitchTalmadgeUofU wonky. Here's the code that fetches and sets the MIME type: https://github.com/webpack/webpack-dev-middleware/blob/9afcd36b55b5c30810d4e9b7e69ba50696de2cd8/lib/middleware.js#L70-L77 I don't get the same error. Mac High Sierra and same build of Chrome here. 

You might have to resort to stepping through the code on your end to see why `mime` is failing there.

**Comment by MitchTalmadgeUofU** [comment]
@shellscape Try it on Windows. [This person](https://github.com/webpack/webpack-dev-server/issues/1373#issuecomment-380189389) had the problem on Windows only.

**Comment by shellscape** [comment]
Well in that case, someone with Windows already installed will have to give some that some debugging love 😅  If a cause can be locked down, we can create a test around it that'll run in AppVeyor to prevent a backslide.

**Comment by GuacheSuede** [comment]
@shellscape Not failing on latestwebpack-dev-middleware builds in regards to `MIME` errors however `Cannot GET /` error comes up thereafter

**Comment by MitchTalmadgeUofU** [comment]
We discovered that the problem is solved when the path does not have spaces in it. `Cannot GET /` goes away and the page is loaded perfectly when the `contentBase` has no spaces.

Good: `contentBase: "C:\Users\mitch\Downloads",`
Bad: `contentBase: "C:\Users\mitch talmadge\Downloads",`

**Comment by MitchTalmadgeUofU** [comment]
I believe this is what is causing the problem: https://github.com/webpack/webpack-dev-middleware/pull/288

I will have to test it.

**Comment by GuacheSuede** [comment]
Thanks for the debugging @MitchTalmadgeUofU .The handicap of not having spaces will cause many incompatibilities with many developers.

**Comment by MitchTalmadge** [comment]
@GuacheSuede No worries!

Does your fully resolved path have spaces in it?

**Comment by GuacheSuede** [comment]
@MitchTalmadge Yes it does.

**Comment by aj-r** [comment]
I think this is a duplicate of #297. It is fixed in `webpack-dev-middleware@3.1.3`

**Comment by shellscape** [comment]
@aj-r indeed, it looks that way. Closing referencing #297.
